sdk/executor/src/main/java/com/mesosphere/sdk/executor/CustomExecutor.java

```diff
@@ -86,21 +88,56 @@ public void launchTask(final ExecutorDriver driver, final Protos.TaskInfo task)
         }
     }
 
-    private void scheduleHealthCheck(Protos.TaskInfo taskInfo, LaunchedTask launchedTask) {
+    private void scheduleHealthCheck(
+            ExecutorDriver executorDriver,
+            Protos.TaskInfo taskInfo,
+            LaunchedTask launchedTask) {
+
         if (!taskInfo.hasHealthCheck()) {
             LOGGER.info("No health check for task: " + taskInfo.getName());
             return;
         }
 
+        scheduleCheck(executorDriver, taskInfo, taskInfo.getHealthCheck(), launchedTask);
+    }
+
+    private void scheduleReadinessCheck(
+            ExecutorDriver executorDriver,
+            Protos.TaskInfo taskInfo,
+            LaunchedTask launchedTask) {
+
+        Optional<Protos.HealthCheck> readinessCheckOptional = Optional.empty();
+        try {
+            readinessCheckOptional = CommonTaskUtils.getReadinessCheck(taskInfo);
+        } catch (TaskException e) {
+            LOGGER.error("Failed to extract readiness check.", e);
+            return;
+        }
+
+        if (!readinessCheckOptional.isPresent()){
+            LOGGER.info("No readiness check for task: " + taskInfo.getName());
+            return;
+        }
+
+        scheduleCheck(executorDriver, taskInfo, readinessCheckOptional.get(), launchedTask);
+    }
+
+    private void scheduleCheck(
+        ExecutorDriver executorDriver,
```

Comment:
Nit, one more level of indentation here so that it's obvious where the method body begins.